viz_scripts/scaffolding.py

```diff
@@ -124,15 +125,23 @@ def load_viz_notebook_data(year, month, program, study_type, dic_re, dic_pur=Non
     # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
     if "distance" in expanded_ct.columns:
         unit_conversions(expanded_ct)
-
-    # Mapping new mode labels with dictionaries
+    
+    # Map new mode labels with translations dictionary from dynamic_labels
     # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
     if "mode_confirm" in expanded_ct.columns:
-        expanded_ct['Mode_confirm']= expanded_ct['mode_confirm'].map(dic_re)
+        if (len(dynamic_labels)):
+            dic_mode_mapping = mapping_labels(dynamic_labels, "MODE")
+            expanded_ct['Mode_confirm'] = expanded_ct['mode_confirm'].map(dic_mode_mapping)
+        else:
+            expanded_ct['Mode_confirm'] = expanded_ct['mode_confirm'].map(dic_re)
```

Comment:
nit: I am fairly sure you can refactor these three copies into one.
I am fine with deferring this to a **future fix**.